server/src/core/file/resolvers/file-upload.resolver.ts

```diff
@@ -0,0 +1,60 @@
+import { Args, Mutation, Resolver } from '@nestjs/graphql';
+import { GraphQLUpload, FileUpload } from 'graphql-upload';
+import { v4 as uuidV4 } from 'uuid';
+import { FileUploadService } from '../services/file-upload.service';
+import { UseGuards } from '@nestjs/common';
+import { JwtAuthGuard } from 'src/guards/jwt.auth.guard';
+import { streamToBuffer } from 'src/utils/stream-to-buffer';
+import { FileFolder } from '../interfaces/file-folder.interface';
+
+@UseGuards(JwtAuthGuard)
+@Resolver()
+export class FileUploadResolver {
+  constructor(private readonly fileUploadService: FileUploadService) {}
+
+  @Mutation(() => String)
+  async uploadFile(
+    @Args({ name: 'file', type: () => GraphQLUpload })
+    { createReadStream, filename, mimetype }: FileUpload,
+    @Args('fileFolder', { type: () => FileFolder, nullable: true })
+    fileFolder: FileFolder,
+  ): Promise<string> {
+    const stream = createReadStream();
+    const buffer = await streamToBuffer(stream);
+    const ext = filename.split('.')?.[1];
```

Comment:
move these 3 lines to a service? not very important but looks like logic we want to re-use

Reply:
I'll definitely take a look on that